scipy/signal/signaltools.py

```diff
@@ -203,6 +213,43 @@ def correlate(in1, in2, mode='full', method='auto'):
     except KeyError:
         raise ValueError("Acceptable mode flags are 'valid',"
                          " 'same', or 'full'.")
+    if return_lags:
+        # cross correlate inputs
+        xcorr = convolve(in1, _reverse_and_conj(in2), mode, method)
+
+        # calculate lag ranges in different modes of operation
+        if mode == "full":
+            # the output is the full discrete linear convolution
+            # of the inputs. (Default)
+            lags = np.arange(-in2.size+1, in1.size)
```

Comment:
This only works, or even makes any sense, for 1D input.

Reply:
Agree that it only makes sense for 1D inputs, but with sufficient docs, is this not okay? 

If you go down the route of adding a new func, it's just going to wrap correlate anyway and it'll make things fragmented.

Reply:
>If you go down the route of adding a new func, it's just going to wrap correlate anyway

I don't see what benefit we get from returning both outputs from the same function. Just have a function `correlation_offsets` (or so) that returns _only_ the offsets.